### PR TITLE
Add variable binding expression example

### PR DESCRIPTION
**Pull request by malwoodsantoro**
This example uses `let` and `var` expressions to demonstrate variable binding syntax. To test this out locally, you'll need to replace the url to the GeoJSON with the raw data. 

:)

### CODE REVIEW

File: docs/pages/example/visualize-population-density.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/*---
+title: Visualize population density
+description: Using a [`varible binding expression`](/mapbox-gl-js/style-spec#expressions-variable-binding) to calculate and display population density.
```
Comment (ryanhamley) [inline docs/pages/example/visualize-population-density.js]: nit: `variable` is misspelled here

File: docs/pages/example/visualize-population-density.html [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                        1551,
+                        ["to-color", "#006d2c"]
+                    ],
+                    10,
+                    [
+                        "interpolate",
+                        ["linear"],
+                        ["var", "density"],
+                        274,
+                        ["to-color", "#eff3ff"],
```
Comment (ryanhamley) [inline docs/pages/example/visualize-population-density.html]: I *think* @ChrisLoer 's recent work makes the explicit cast to a color here unnecessary

Comment (ChrisLoer) [reply docs/pages/example/visualize-population-density.html]: I think the cast is necessary until https://github.com/mapbox/mapbox-gl-js/pull/7301 has merged and is being used by the example page, right?

Comment (ryanhamley) [reply docs/pages/example/visualize-population-density.html]: LOL, oh yeah. I was working on that. Fridays.

File: docs/pages/example/visualize-population-density.html [hunk 3]
```diff
@@ -0,0 +1,53 @@
+<div id='map'></div>
+<script>
+var map = new mapboxgl.Map({
+    container: 'map', // container id
+    style: 'mapbox://styles/mapbox/streets-v9', // stylesheet location
```
Comment (ChrisLoer) [inline docs/pages/example/visualize-population-density.html]: I have a personal vendetta against streets-v9 in examples. The flattened road labels are 🙅‍♂️

**Comment by ChrisLoer** [comment]
> The code is almost identical to our choropleth example but with the added overhead of declaring the variable syntax

🤔 Yeah I see your point although it seems like using `let` and `var` will always be _part_ of accomplishing something else (hopefully with less duplication in the expression code). I guess this seems like a pretty reasonable choice to me.

**Comment by ryanhamley** [comment]
I'm ok with it then. Just a couple little nits to clean up. I'll update the example to remove the explicit casts to color when I finish up the test in https://github.com/mapbox/mapbox-gl-js/pull/7301

**Pull request merged**